crates/subgraph/src/utils/mod.rs

```diff
@@ -1,5 +1,79 @@
+use alloy::primitives::{
+    utils::{format_units, parse_units, ParseUnits, Unit, UnitsError},
+    I256, U256,
+};
+use chrono::TimeDelta;
+
 mod order_id;
 mod slice_list;
 
 pub use order_id::*;
 pub use slice_list::*;
+
+/// Returns 18 point decimals 1 as I256/U256
+pub fn one_18() -> ParseUnits {
+    ParseUnits::U256(U256::from(1_000_000_000_000_000_000_u64))
+}
+
+/// Returns YEAR as 18 point decimals as I256/U256
+pub fn year_18() -> ParseUnits {
+    ParseUnits::U256(
+        U256::from(TimeDelta::days(365).num_seconds()).saturating_mul(one_18().get_absolute()),
+    )
+}
+
+/// Converts a U256/I256 value to a 18 fixed point U256/I256 given the decimals point
+pub fn to_18_decimals<T: TryInto<Unit, Error = UnitsError>>(
+    amount: ParseUnits,
+    decimals: T,
+) -> Result<ParseUnits, UnitsError> {
+    parse_units(&format_units(amount, decimals)?, 18)
+}
+
+/// Returns annual rate as 18 point decimals as I256
+pub fn annual_rate(start: u64, end: u64) -> I256 {
```

Comment:
can this function just take two 18 decimal values and calculate the rate?

the way it is now, it complects a bunch of stuff

- converting u64 to i256
- rescaling decimal 0 to decimal 18
- ad hoc inlined 18 decimal devision